### PR TITLE
Atomics support in interpreter + fuzz fixes for that

### PR DESCRIPTION
**Pull request by kripken**
After this, the fuzzer seems to not find anything. So at least the interpreter appears consistent with itself ;)

### CODE REVIEW

**Comment by kripken** [comment]
Merging as this blocks other stuff. Feedback still welcome of course.

File: src/ast/bits.h [hunk 1]
```diff
@@ -60,6 +62,43 @@ struct Bits {
     }
     WASM_UNREACHABLE();
   }
+
+  static Expression* makeSignExt(Expression* value, Index bytes, Module& wasm) {
```
Comment (binji) [inline src/ast/bits.h]: The threads proposal has new instructions for this: https://github.com/WebAssembly/threads/blob/master/proposals/threads/Overview.md#new-sign-extending-operators

Comment (kripken) [reply src/ast/bits.h]: Interesting. I guess when we add those to binaryen we can use them to replace `makeSignExt` etc.

Comment (dschuff) [reply src/ast/bits.h]: Binaryen already has them: https://github.com/WebAssembly/binaryen/pull/1167

**Pull request merged**